modules/doi/includes/utilities.inc

```diff
@@ -204,12 +205,9 @@ function islandora_doi_crossref_translator($crossref_data) {
           case 'full_title':
             $titleinfo = $mods->createElement('titleInfo');
             $relateditem->appendChild($titleinfo);
-            $title = $mods->createElement('title');
+            $title_text = islandora_doi_filter_html($crossref_xml->saveXML($child->firstChild));
```

Comment:
Not sure, but should we not keep testing for `$child->firstChild` being not `NULL` (here and elsewhere)? Or does `DOMDocument::saveXML()` behave well in this regard?

Reply:
My testing didn't throw any issues with DOIs with no subtitles even though we're doing the same thing with those, but then again it may be including empty subtitle elements that are getting stripped by Islandora's default MODS cleanup script, but I suppose better safe than sorry. I'll add some NULL tests.